src/tests/unit_tls.cpp

```diff
@@ -863,6 +991,65 @@ class TLS_Unit_Tests final : public Test {
                        Botan::TLS::Alert::InternalError);
       }
 
+      void test_custom_ecdh_provider(std::vector<Test::Result>& results,
+                                     const std::shared_ptr<Credentials_Manager_Test>& creds,
+                                     const std::shared_ptr<Botan::RandomNumberGenerator>& rng) {
+         auto noop_session_manager = std::make_shared<Botan::TLS::Session_Manager_Noop>();
+
+         const auto groups = {
+            Botan::TLS::Group_Params::SECP256R1,
+            Botan::TLS::Group_Params::BRAINPOOL256R1,
+            Botan::TLS::Group_Params::BRAINPOOL512R1,
+         };
```

Comment:
What about X25519 and X448 keys?

Reply:
Those need special handling in the callbacks, because they are not used via their `ECDH_PrivateKey` but have their own implementations. Albeit, you're right. We should test those as well.